runtime/parachains/src/paras_inherent.rs

```diff
@@ -349,20 +349,39 @@ pub mod pallet {
 
 			let max_block_weight = <T as frame_system::Config>::BlockWeights::get().max_block;
 
-			// Abort if the total weight of the block exceeds the max block weight
-			if candidate_weight.saturating_add(bitfields_weight).saturating_add(disputes_weight) > max_block_weight {
-				backed_candidates.clear();
-				candidate_weight = 0;
-				signed_bitfields.clear();
-				bitfields_weight = 0;
-			}
+			// Potentially trim inherent data to ensure processing will be within weight limits
+			let total_weight = {
```

Comment:
hopefully not controversial, i just thought it was easier to understand when we grouped this logic into a block

Reply:
If it survives `cargo +nightly fmt`, :+1: